### PR TITLE
feat: iOS - Add SPM support

### LINKED ISSUE #566: iOS - try to solve the way we use YogaKit as a dependency both with CocoaPods and inside our repo
## Problems

- when using Cocoapods, users need to specify a dependency to a Yoga fork `pod 'YogaKit', :git => 'https://github.com/lucasaraujo/yoga.git'` which is owned by @lucasaraujo. Besides being weird, that can lead to inconsistency between versions of Yoga.

- We currently have all source code of Yoga inside our monorepo. Depending on *how* someone would use Beagle, this could lead to inconsistencies:
  - CocoaPods: users will get source code outside our monorepo
  - Carthage: users will get the exact source code from our monorepo
  - Internal usage: internal developers and CI use source code from monorepo

## Expected

The best solution was to rely on exactly the same source code and version of YogaKit for any of the methods described above. But a further investigation is needed to properly solve this issue

### LINKED ISSUE #692: iOS - Provide integration through Swift Package Manager
## Description

Today Beagle offers integrations through `Carthage` and `CocoaPods`, however with the growth in the use of the `SPM` (`Swift Package Manager`) by the community it would be interesting to offer integration through this dependency management tool.

## Acceptance Criteria

A developer being capable of using `SPM` to add Beagle to a Xcode project

### PR DESCRIPTION
**Pull request by dantes-git**
### Related Issues

#566, Closes #692

### Description and Example

This PR adds support for SPM in Beagle iOS as well as other requirements for this feature:

- [x]  Removes YogaKit sources from Beagle
- [x]  Refactor Test folder structure (separates all tests from library sources)
- [x]  Configures Package.swift

Enhancements:

- [x]  Creates a sample project running local for ci checks
- [ ]  Updates YogaKit version on Package.swift after the release
- [ ]  Configures BeaglePreview in SPM
- [ ]  Adds documentation

You can test this branch by adding a new SPM dependency in you project:

Captura de Tela 2021-04-12 às 17 26 00

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// swift-tools-version:5.1
```
Comment (igorgiumellizup) [inline Package.swift:1]: Shouldn't it contain the Zup header?

**Review by luisgustavozup — DISMISSED** [review]

File: Package.swift (hunk 1 above)
Comment (dantes-git) [reply Package.swift:1]: Yes, thank you, I was forgetting

File: fastlane/Fastfile [hunk 2]
```diff
@@ -97,6 +98,15 @@ platform :ios do
     )
   end
 
+  desc "Check wether Beagle is working via SPM in an Example Project"
```
Comment (igorgiumellizup) [inline fastlane/Fastfile:101]:
```suggestion
  desc "Check whether Beagle is working via SPM in an Example Project"
```

**Review by lucasaraujo — DISMISSED** [review]

**Review by lucasaraujo — APPROVED** [review]

**Review by luisgustavozup — APPROVED** [review]

**Pull request merged**